ts/private/ts_project_worker.js

```diff
@@ -222,11 +225,9 @@ function createProgram(args, initialInputs) {
 
         if (kind === ts.FileWatcherEventKind.Created) {
             knownInputs.add(filePath);
-
-            const dirname = path.dirname(filePath);
-            getDirectoryWatcherForPath(filePath)?.(dirname);
-
-            // signal that the directory has been created.
+            const directoryWatcher = getDirectoryWatcherForPath(filePath)
+            directoryWatcher?.(path.dirname(filePath));
+            directoryWatcher?.(filePath);
```

Comment:
Why do we need to invoke the directory watcher with both the directory and file?

Reply:
to signal that there have been some changes in that directory.

Reply:
We have to signal it twice though? Does that fix something?

Reply:
given filePath `/tmp/bin/feat/1.ts`  first we have to signal there is been some changes in `/tmp/bin/feat` which is what line 229 does by getting dirname from filepath. the second one actually specifies which file has changed.

Reply:
Is that part of a contract or API that we are following? Why do we have to do that?

Reply:
Yes without this logic tsc does not pick up changes properly.

Reply:
As long as there's a test that fails when either of those two lines is removed. Or a link to something explaining it. Is this a typescript API with docs somewhere saying this?

Reply:
I couldn't come across a proper implementation `ts.sys`. my assumption was that it's how a real fs would behave when tsc calls `watchDirectory` with `recursive = true`

a simple example 

```ts
const ts = require("typescript");
ts.sys.watchDirectory("./dir", console.log, true);
```

will yield 

```bash
dir/bin
dir/bin/1.ts
```

adding a test for this is gonna be a tricky one but I'll follow with another PR